### ISSUE TITLE
Internet connection lost after a few minutes of use in 0.10.0

### ISSUE THREAD

**Issue by orbitz** [pgj/freebsd-wifibox#28]
I've upgraded wifibox to the following versions:

```
Installed packages to be UPGRADED:
        wifibox-alpine-ath10k: 20220430 -> 20220625
        wifibox-ath10k: 1.0.0 -> 1.1.0
        wifibox-core: 0.9.0 -> 0.10.0

```

After a few minutes of internet use, hosts stop resolving and I cannot contact any hosts.

Additionally, `wpa_cli status` times out.

I'm on:

```
FreeBSD bsdell 14.0-CURRENT FreeBSD 14.0-CURRENT #71 main-n256742-db0110a5
```

I've reverted and things seem to be working as expected.

Let me know if there is any other information that would help.

**Comment by pgj** [comment]
Hi @orbitz: it would be definitely good to see some logs, especially around from the time when things go wrong.  This might be related to the amount of memory assigned for the guest.  In version 1.1.0, this is 45M for all the flavors, but I have just noticed this week that it may only reliably apply for `iwlwifi` (the one that I use regularly), other flavors, such as `ath10k` may need more than that.

There is a fix in preparation for this, version 1.1.1, that you can find here in the meantime: https://github.com/pgj/freebsd-wifibox-port/tree/13df793ea46725e43c5c35292286c2f2e4b18195 -- could you please take a try with this?

**Comment by orbitz** [comment]
Quick question, not quite related: is there an easy way to rollback to a previous install after testing?  I'm just getting lucky and the previous wifibox versions are in my pkg cache.

**Comment by pgj** [comment]
You can always use the port.  The `freebsd-wifibox-port` repository has [all the releases tagged](https://github.com/pgj/freebsd-wifibox-port/releases).

**Comment by orbitz** [comment]
How do I install via port when I get this error?

> Installing wifibox-alpine-20220712...
pkg-static: wifibox-alpine-20220712 conflicts with wifibox-core-0.9.0 (installs files into the same place).  Problematic file: /usr/local/etc/wifibox/wpa_supplicant/wpa_supplicant.conf.sample

**Comment by orbitz** [comment]
Just because I'm not familiar much with building ports like this, I would like to confirm what to do:

Should I just build wifibox-alpine and wifibox-core and install those?  The wifibox one is a metaport, I take it I do not need to install that?  Currently on my system I have the following ports installed, should I uninstall them before doing this?

```
wifibox-alpine-ath10k-20220430
wifibox-ath10k-1.0.0
wifibox-core-0.9.0
```

**Comment by pgj** [comment]
Yes, in practice, only `net/wifibox-alpine` and `net/wifibox-core` what matter.  The `net/wifibox` port has become a metaport and its purpose is to lock the properly matching versions of these other ports.  If you do not install ports of conflicting versions, the metaport is not needed.  (I do not use the metaport itself that much because I am mindful about the version requirements, but do not try this at home.)  Also, the metaport is to make it easier to install the entire suite from package.

My usual strategy to move between versions is as follows.

1. Update the `freebsd-wifibox-ports` git repository.
2. As a regular user, do `make` for both `net/wifbox-core` and `net/wifibox-alpine`.  Note that there you can set the flavor for `net/wifibox-alpine`, i.e. `FLAVOR=ath10k` in your case.  Missing out on this is not a big problem, you will get a bit fatter image in result but that is all.  It is important _not to_ `install` anything yet.  Reason: you cannot install the port due to the conflicting package and you will need a working Internet connection for the time of fetching the source files.
3. `pkg delete wifibox-core wifibox-alpine-ath10k`.
4. Run `make install` as `root` for both `net/wifibox-core` and `net/wifibox-alpine` (+ `FLAVOR=ath10k`) to install the ports one by one.  There is no strict dependency between each other, hence they can be installed in any order.
5. If everything looks fine, `make clean` for the ports and enjoy.

Please note that [`net/wifibox-core` 0.10 has brought some breaking changes over 0.9](https://github.com/pgj/freebsd-wifibox/releases/tag/0.10.0), be careful with them.   Make sure you read and understand the [`freebsd-wifibox-port` `README`](https://github.com/pgj/freebsd-wifibox-port#readme).

**Comment by orbitz** [comment]
Good news, I'm on the commit you pointed me at and, at least so far, still going good.  I will keep going on this and let you know.  But previously I did not make it as long a I have now.

**Comment by pgj** [comment]
@orbitz I assume that version 1.1.1. solves this problem.  Note that the FreeBSD ports tree has been updated to that version and packages will be coming soon.

**Comment by fearedbliss** [comment]
I'm also experiencing this issue on the framework laptop (using wifibox 1.1.1). The internet could work for a non-deterministic amount of time and then it would drop. Once it drops, it will continue dropping every few minutes (I need to do `wifibox restart && dhclient wifibox0` to get internet again). A hard reboot helps it "last longer", but it will eventually die again. I'm not experiencing this issue on my other Thinkpad X1C7 that had the bhyve pci drop issue.

**Comment by pgj** [comment]
The root cause might be the same, i.e. the guest memory is too low to run reliably.  It is 45 MB in case of `iwlwifi`, which works for me, but as a test, you can try increasing it by changing the `memory` value in `$SHAREDIR/wifibox/bhyve.conf`.

**Comment by fearedbliss** [comment]
@pgj The defaults for my bhyve.conf file says 128M. I haven't changed it so that's what was placed when I installed the `wifibox` package. I'll try 512M just to see if a more extreme value helps

**Comment by pgj** [comment]
The configuration file I was talking about in my comment above is guest-specific and it is under the `$LOCALBASE/share/wifibox` directory.  It can override whatever the user specifies.  Although this might not be the best design choice, that is how it works at the moment.

**Comment by fearedbliss** [comment]
@pgj Ah! That's for clarifying. I'll look at that file then ;).

**Comment by fearedbliss** [comment]
At the default of 45M (in localshare):

the `free -m` in the vm said:

```
wifibox:~# free -m
              total        used        free      shared  buff/cache   available
Mem:             27          23           2           0           3           2
Swap:             0           0           0
```

after I switched it to 512M it said:

```
wifibox:~# free -m
              total        used        free      shared  buff/cache   available
Mem:            487          24         448           0          15         451
Swap:             0           0           0
```

Will let you know if it fixes it. Thanks for the help :D.

**Comment by fearedbliss** [comment]
So far so good, I'll still need more time to test since I didn't use my computer that much yesterday. I think it would be good to remove the **/usr/share/local/** version of bhyve.conf and exclusively use the one in the **/usr/local/etc/wifibox/** directory. This simplifies debugging and increases predictability since that would be the only value where memory could be controlled. It may even also help with future wifibox upgrades since configuration files in the local etc dir will most likely not get modified. Should I open another ticket for that?

**Comment by pgj** [comment]
The reason is that `$LOCALBASE/share/wifibox` is controlled by the guest which is installed independently.  With a `bhyve.conf` deployed there, the guest itself has the chance to set the optimal amount of memory for itself.  This is a value that the actual composition of the guest system determines and it should not be something that the user has to figure out.  For example, the `iwlwifi` driver can work with only 45 MB of RAM (for me, at least) but the `ath10k` driver needs ~60 MB, while both of them based on the same Alpine Linux base.  Leaving it at 128 MB in general sounds like a waste.

I am open to redesigning this feature in a different ticket.  I would like to keep this possibility around but I understand that it is an unwanted complexity from the side of user experience.

**Comment by fearedbliss** [comment]
That makes sense. I would say we should try and have a safe consistent default that would work with no issues for the majority of people without addition configuration (Meaning we should use a bigger value like 128 MB even if it is "wasting" around 60+ MB). The user can always adjust it down at their own discretion. This would also mean we would also want just one configuration file if possible. If someone makes a regular VM, they usually specify exactly what the available memory will be without letting the VM trying to auto optimize down. You could also look at it kinda like a GENERIC kernel, you could argue that a generic kernel is a big waste for basically everyone since nobody has every driver used in the kernel (even if it's modular), but the convenience and consistency gained from having a big GENERIC config that works for everyone without thinking is a bit usability improvement.

**Comment by orbitz** [comment]
This happened to me again.  I just left my machine doing nothing and a few hours later came back to a dead wifibox.  I'll up the memory amount and see what happens.

**Comment by fearedbliss** [comment]
So far I haven't had any more sudden wifibox deaths after increasing the memory. I want to test for a bit longer and will report back again. But so far it's promising.

**Comment by fearedbliss** [comment]
I'm happy to report back that I haven't had any more issues since increasing the memory.
